onboarding.md

```diff
@@ -0,0 +1,591 @@
+  an `EdcException`
+- configuration values should be handled in the `*Extension` class, as it's job is to set up the extension and its
+  required business logic (e.g. the controller). The extension itself should not contain any business logic.
+- it's better to pass the config value directly into the business logic than passing the
+  entire `ServiceExtensionContext`
+
+> _The complete sample code for this chapter is in `samples/03-configuration`._
+
+## Implement a simple file transfer
+
+Now this is a big one:
+
+- We'll create an additional connector, so that in the end we have two connectors, a consumer and a provider.
+- The connectors will talk to each other via IDS, and the consumer will expose a REST API so that external systems (e.g.
+  users) can interact with them.
+- The consumer will initiate a file transfer and the provider will fulfill that request and copy a file to the desired
+  location.
+- Both connectors will run locally on the development machine
+
+Also, in order to keep things organized, the code in this example has been separated into several Java modules:
+
+- `[consumer|provider]`: this is where the "main" classes and configs reside for both the consumer and the provider
+  connector
+- `api`: contains the REST API extension (previously named `HealthApiExtension`)
+- `transfer-file`: contains all the code necessary for the file transfer
+
+### Create a control REST API
+
+We will need some way to interact with the consumer, a communication protocol of sorts. That is what we call
+outward-facing communication. In this example we will communicate with the consumer via simple command line tools
+like `cURL`, but it is easy to imagine some other much more complicated control system to interact with the (consumer)
+connector. Thus, using Jakarta, we must create an API controller the same way we created our health endpoint a few
+chapters back. In fact, we can re-use and
+improve [that controller](samples/04-file-transfer/api/src/main/java/org/eclipse/dataspaceconnector/extensions/api/ConsumerApiController.java)
+(code omitted here for brevity).
+
+Again, like before, the controller is instantiated and registered in an extension which we aptly
+name `ApiEndpointExtension.java`.
+
+### Create the "provider" connector
+
+The provider is the one "owning" the data. That means, the consumer sends a request to the provider (over IDS), who then
+performs the file transfer. The provider must maintain an internal list of assets that are available for transfer, the
+so-called "catalog". For the sake of simplicity we use an in-memory catalog and pre-fill it with just one single class:
+
+```java
+// in FileTransferExtension.java
+@Override
+public void initialize(ServiceExtensionContext context){
+        // ...
+        registerDataEntries(context);
+        // ...
+        }
+
+private void registerDataEntries(ServiceExtensionContext context){
```

Comment:
wonder if we should mention that this code will most likely change in the upcoming weeks as we want to integrate AssetIndex, ContractOffers etc and replace the MetadataStore as it is

Reply:
lets just add a general "Code is under development" remark somewhere prominent in the file